### DISCUSSION TITLE
[Bug]: Sawmill acceptance limitations

### DISCUSSION THREAD

**Discussion by Hevlikn** [OpenTTD/OpenTTD#11732] [Features]
### Version of OpenTTD

12.1

### Expected result

Any tile of an industry should have full acceptance/generation, for consistency. Currently, mines, forests, factories, farms, and steel mills follow this behaviour.

### Actual result

Sawmills can only accept wood if the middle south-western tile is covered by a freight-type station (ie not bus or passenger tram).

I believe this is a simplification to account for the variants of the industries, as there are both 3x3 sawmills and 'corner cut' sawmill types, while other industries exhibiting the desired behaviour are simple rotations, or variants have the same footprint.

As an aside, power stations only accept from the central 4 tiles, and this would further confirm the above theory, with the 2x4 'long' variant, and the 3x3 -1 'cut corner' variant having only 4 accepting tiles.

While simply to avoid for a veteran player looking at acceptance/supply while placing, this inconsistent behaviour [is illogical for newer players 
](https://old.reddit.com/r/openttd/comments/s6gm6i/why_isnt_it_accepting_timber_please/)

### Steps to reproduce

https://imgur.com/a/2UH1mVu

Initially detected in JGRPP, but confirmed in 12.1. No GRFs. 

Can be confirmed from new game (no older saves compatible to test against) - not a 'delayed demand from new industry' problem.

Searches `is:issue sawmill coverage`, `is:issue sawmill acceptance` `is:issue sawmill wood` returned no relevant results

**Comment by Hevlikn** [comment]
added the 3 power station variants to the above album, looks like it is either 1 or 2 tiles rather than 4.

Running through other industries;

Temperate:
    Oil refineries
    Power Stations
    Sawmills
    
Arctic:
    Oil refineries
    Power Stations
    
Desert:
    Oil Refineries
    
(Of course Candyland is unaffected!)

**Comment by glx22** [comment]
This is intended behaviour, it comes from the original TTD.
Each tile has an acceptance of x/8, and the cargo is accepted if the sum of is >1 in station coverage.
Land inspection tool can show it.
Same happens for buildings in town (for passengers, mail and goods in vanilla).

**Comment by Hevlikn** [comment]
> Each tile has an acceptance of x/8, and the cargo is accepted if the sum of is >1 in station coverage.

~~This does not explain the behaviour of the other industries, and is even inconsistent with itself as a forest can supply wood from *any* tile.

Additionally, this does not explain why covering the 6 other tiles would not generate demand~~


Ah i misunderstood; yes that is correct, but it remains that it is inconsistent and unintuitive.

**Comment by 2TallTyler** [comment]
For what it's worth, I don't think partial tile acceptance for industries is necessarily a useful, fun feature which enhances the game. I wouldn't be sad to see it disappear and have full 8/8 acceptance on every vanilla industry tile.

**Comment by TrueBrain** [comment]
I feel a new setting coming up .. or we can skip the setting part ofc. But yes, it is one of those design decisions that isn't particular good :P

Marking this as feature-request for now, as it isn't actually a bug. But I think many more people agree it is a bad concept :)
